src/server/gc_worker/gc_worker.rs

```diff
@@ -367,18 +367,28 @@ where
         MvccTxn::new(TimeStamp::zero(), concurrency_manager)
     }
 
-    fn flush_txn(txn: MvccTxn, limiter: &Limiter, engine: &E) -> Result<()> {
-        let write_size = txn.write_size();
-        let modifies = txn.into_modifies();
-        if !modifies.is_empty() {
+    fn flush_txn(
+        limiter: &Limiter,
+        engine: &E,
+        txns: std::collections::HashMap<u64, MvccTxn>, // region id -> MvccTxn for this region
+    ) -> Result<()> {
+        let mut modifies = std::collections::HashMap::new();
+        let mut write_size = 0;
+        for (id, txn) in txns {
+            write_size += txn.write_size();
+            modifies.insert(id, txn.into_modifies());
+        }
+        if write_size > 0 {
             limiter.blocking_consume(write_size);
             engine.modify_on_kv_engine(modifies)?;
         }
         Ok(())
     }
 
     fn gc(&mut self, region: Region, safe_point: TimeStamp) -> Result<()> {
-        if !self.need_gc(region.get_start_key(), region.get_end_key(), safe_point) {
+        if !self.engine.multi_rocks()
```

Comment:
Why need to check if it's multi rocks?

Reply:
multi rocks has not this engine for this check.